isthmus/src/main/java/io/substrait/isthmus/AggregateValidator.java

```diff
@@ -0,0 +1,223 @@
+package io.substrait.isthmus;
+
+import io.substrait.expression.AggregateFunctionInvocation;
+import io.substrait.expression.Expression;
+import io.substrait.expression.FieldReference;
+import io.substrait.expression.FunctionArg;
+import io.substrait.expression.ImmutableExpression;
+import io.substrait.expression.ImmutableFieldReference;
+import io.substrait.relation.Aggregate;
+import io.substrait.relation.Project;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * Not all Substrait {@link Aggregate} rels are convertable to {@link
+ * org.apache.calcite.rel.core.Aggregate} rels
+ *
+ * <p>The code in this class can:
+ *
+ * <ul>
+ *   <li>Check for these cases
+ *   <li>Rewrite the Substrait {@link Aggregate} such that it can be converted to Calcite
+ * </ul>
+ */
+public class AggregateValidator {
```

Comment:
I'm open to suggestions for a better name for this.

Reply:
This inspired me to use `PreCalciteAggregateValidator` and `PreCalciteAggregateTransformer`.

It's very similar to `AggregatePreCalciteTransformer` but with extra emphasis on the `PreCalcite` part of it.